.github/workflows/test.yml

```diff
@@ -124,28 +124,50 @@ jobs:
         continue-on-error: true
         uses: AndreMiras/coveralls-python-action@v20201129
 
-  release:
+  build_sdist:
+    needs: pytest
+    name: Build source distribution
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        name: Install Python
+        with:
+          python-version: '3.10'
+      - run: python -m pip install build
+      - name: Build sdist
+        run: python -m build --sdist
+      - uses: actions/upload-artifact@v3
+        with:
+          path: dist/*.tar.gz
+
+  build_wheels:
     needs: pytest
-    if: github.event_name == 'release' || (github.event_name == 'workflow_dispatch' && inputs.task == 'release')
     strategy:
       matrix:
-        os: [macos-latest, windows-latest]
-        python-version: ['3.8', '3.9', '3.10', '3.11']
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ['38', '39', '310', '311']
     runs-on: ${{ matrix.os }}
-
     steps:
       - uses: actions/checkout@v3
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
-        with:
-          python-version: ${{ matrix.python-version }}
-          cache: pip
-          cache-dependency-path: setup.py
-      - name: Release
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.11.4
         env:
-          TWINE_USERNAME: ${{ secrets.TWINE_USERNAME }}
-          TWINE_PASSWORD: ${{ secrets.TWINE_PASSWORD }}
-        run: |
-          python -m pip install --upgrade pip twine build
-          python -m build
-          python -m twine upload --skip-existing dist/*.whl dist/*.tar.gz
+          CIBW_BUILD: cp${{ matrix.python-version }}-*
+      - uses: actions/upload-artifact@v3
+        with:
+          path: ./wheelhouse/*.whl
+
+  release:
+    needs: [build_wheels, build_sdist]
+    if: github.event_name == 'release' || (github.event_name == 'workflow_dispatch' && inputs.task == 'release')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/download-artifact@v3
+        with:
+          name: artifact
+          path: dist
+      - uses: pypa/gh-action-pypi-publish@master
+        with:
+          user: ${{ secrets.TWINE_USERNAME }}
+          password: ${{ secrets.TWINE_PASSWORD }}
```

Comment:
Does this offer advantages over running `twine upload` manually? I'm wondering if it's worth relying on an external action for sth this simple since it might lock us into waiting for fixes if the action ever breaks.

Reply:
Ok, `twine upload` should also work.